File: 3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.cpp

class Solution {
public:
    int valueAfterKSeconds(int n, int k) {
        const int MOD = 1000000007;
        vector<int> arr(n, 1);

        for (int step = 0; step < k; step++) {
            for (int i = 1; i < n; i++) {
                arr[i] = (arr[i] + arr[i - 1]) % MOD;
            }
        }

        return arr[n - 1];
    }
};

File: 3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number}
 */
var valueAfterKSeconds = function(n, k) {
    const MOD = 1000000007;

    let arr = new Array(n).fill(1);

    for (let step = 0; step < k; step++) {
        for (let i = 1; i < n; i++) {
            arr[i] = (arr[i] + arr[i - 1]) % MOD;
        }
    }

    return arr[n - 1];
};